lib/logstash/inputs/http.rb

```diff
@@ -177,12 +182,35 @@ def decode_body(headers, remote_address, body, default_codec, additional_codecs)
   end
 
   def push_decoded_event(headers, remote_address, event)
+    add_ecs_fields(headers, event)
     event.set(@request_headers_target_field, headers)
     event.set(@remote_host_target_field, remote_address)
     decorate(event)
     @queue << event
   end
 
+  def add_ecs_fields(headers, event)
+    return if ecs_compatibility == :disabled
+
+    http_version = headers.get("http_version")
+    http_user_agent = headers.get("http_user_agent")
+    http_host = headers.get("http_host")
+    request_method = headers.get("request_method")
+    request_path = headers.get("request_path")
+    content_length = headers.get("content_length")
+    content_type = headers.get("content_type")
+
+    event.set("[http][version]", http_version) if http_version
+    event.set("[user_agent][original]", http_user_agent) if http_user_agent
+    domain, colon, port = http_host.rpartition(":")
```

Comment:
not sure this will work as expected e.g.
```
>> 'localhost'.rpartition(":")
=> ["", "", "localhost"]
```
... think we'll have the `host(:port)` part here but feel free to correct me if this returns the raw `Host: host(:port)`,
in either case I think there are cases where it won't work (I am only seeing a test for the `host:port` case maybe having a test for `host` would be useful).

Reply:
you are right. I thought it returns the default 80 port but it didn't. 
I added a regex to handle `ipv4:port` and `[ipv6]:port`

Reply:
Nice catch 🐞

Reply:
thanks, the new code looks much better.
in terms of the 80 default sorry to cause confusion, I do not think we should set that for 2 reasons:
- if the schema is *https* 80 default would be incorrect
- ecs does not require us to set default values that aren't present

so I think having a `nil` default and not setting `url.port` if it wasn't specified would be better for starters.
at least that's the way LS currently parses URLs into `url.*` with grok patterns.

minor thing to consider would be type-casting port to `Integer` but it's not strictly required, 
however we type-cast `url.port` and others with grok.